src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -172,9 +183,39 @@ public async Task DeleteFileIfExistsAsync(long version, string partitionName, Ca
         BlockBlobClient blobClient = GetInstanceBlockBlobClient(version, partitionName);
         _logger.LogInformation("Trying to delete DICOM instance file with watermark '{Version}'.", version);
 
+        EmitTelemetry("DeleteFileIfExistsAsync");
+
         await ExecuteAsync(() => blobClient.DeleteIfExistsAsync(DeleteSnapshotsOption.IncludeSnapshots, conditions: null, cancellationToken));
     }
 
+    private static readonly Action<ILogger, string, Exception> LogBlobClientOperationDelegate =
```

Comment:
Why this pattern? we wanted to avoid this. You can just log in the method itself